util/profiler.go

```diff
@@ -0,0 +1,166 @@
+// Provides continuous profiling for GoLang services in Datadog.
+package util
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/viper"
+
+	"go.uber.org/zap"
+	"gopkg.in/DataDog/dd-trace-go.v1/profiler"
+)
+
+const jitterMaxSec = 60
+
+const (
+	errViperIsNil        = "viper instance is nil"
+	errEnvironmentNotSet = "Environment not set"
+	errVersionNameNotSet = "VersionName not set"
+	errProjectNameNotSet = "ProjectName not set"
+	errServiceNameNotSet = "ServiceName not set"
+)
+
+// See the datadog profiler source for documentation on how these are used
+// dd-trace-go.v1@v1.25.0/profiler/options.go
+type DatadogProfilerConfig struct {
+	Enabled     bool
+	ServiceName string
+	VersionName string
+	ProjectName string
+	Environment string
+}
+
+// Helper function to load config with viper using some standard env vars. This assumes that Viper
+// has already been initialized by the caller.
+// It is recommended to use this directly unless there is a good reason to override the configs.
+func getDatadogProfilerConfigWithViper(v *viper.Viper) (*DatadogProfilerConfig, error) {
+	if v == nil {
+		return nil, errors.New(errViperIsNil)
+	}
+
+	return &DatadogProfilerConfig{
+		Enabled:     v.GetBool("datadog.profiler_enabled"),
+		VersionName: v.GetString("codeflow_release_id"),
+		ProjectName: v.GetString("codeflow.project_name"),
+		Environment: v.GetString("codeflow.config_name"),
```

Comment:
There's a ton of Coinbase-specific references here -- we're going to have to think about how to make it more flexible. Initial thoughts are to continue to leverage flags again or use S3.